lib/ansible/modules/net_tools/dnsmadeeasy.py

```diff
@@ -623,35 +622,49 @@ def main():
         new_record["port"] = new_record["value"].split(" ")[2]
         new_record["value"] = new_record["value"].split(" ")[3]
 
+    # monitor creation
+    monitor_changed = False
+    new_monitor = dict()
+    # if monitor param is passed, we need to build object with necessary parameters
+    if module.params["monitor"] is not None:
```

Comment:
`monitor` is a boolean, so it will never ever be `None`. It will be `False` (by default) or `True` (if the user explicitly sets it to).

Reply:
I thought I was checking, if the monitor param is even set or not ... I need to run the code that follows if monitor parameter is passed (either true or false).

Reply:
A module option with `default` will always be set (to the default value if the user didn't specify anything). You can't detect whether the user actually passed something, except by setting the default to `None` (i.e. omitting default).

Reply:
@felixfontein I have chosen to check for the record_type parameter. That is a lot cleaner. Thank you for pointing this out.